### PR TITLE
RND-18911 - Update WS auth method.

### PR DESCRIPTION
**Pull request by elad-shtilerman**

### CODE REVIEW

File: examples/example_client.py [hunk 1]
```diff
@@ -11,8 +11,9 @@
 # constants
 CHUNK_DURATION_SECONDS = 0.1
 
-# set access token
-streaming_access_token = '<your_access_token>'
+# set token and url
+example_customer_token = '<your_customer_token>'
+example_ws_url = '<websocket url>'  # includes session token in url.
```
Comment (EliAsor) [inline examples/example_client.py:16]:
```suggestion
example_ws_url = '<websocket_url>'  # includes session token in url.
```

Comment (EliAsor) [reply examples/example_client.py:16]: just for uniformity

File: verbit/streaming_client.py [hunk 2]
```diff
@@ -52,11 +54,11 @@ class WebsocketStreamingClientSingleConnection:
...
     NO_RETRY_HTTP_SERVER_CODES = (501, 505, 506, 507, 508, 510)
 
-    def __init__(self, access_token, on_media_error: typing.Callable[[Exception], None] = None):
+    def __init__(self, customer_token, on_media_error: typing.Callable[[Exception], None] = None):
 
         # assert arguments
-        if not access_token:
-            raise ValueError("Parameter 'access_token' is required")
+        if not customer_token:
+            raise ValueError("Parameter 'api_key' is required")
```
Comment (EliAsor) [inline verbit/streaming_client.py:61]:
```suggestion
            raise ValueError("Parameter 'customer_token' is required")
```

File: verbit/streaming_client.py [hunk 3]
```diff
@@ -136,32 +138,36 @@ def socket_timeout(self, timeout: typing.Union[None, float]):
...
     # ========= #
     def start_stream(self,
+                     ws_url: str,
                      media_generator: typing.Iterator[bytes],
                      media_config: MediaConfig = None,
                      response_types: ResponseType = ResponseType.Transcript) -> typing.Iterator[typing.Dict]:
         """
         Start streaming media and get back speech recognition responses from server.
 
+        :param ws_url: websocket url to use.
```
Comment (EliAsor) [inline verbit/streaming_client.py:148]:
```suggestion
        :param ws_url: websocket url to use, as obtained from the Ordering API.
```

Comment (EliAsor) [reply verbit/streaming_client.py:148]: Not sure about the phrasing, but I think it should be clear that the URL needs to be obtained from the Ordering system.

File: verbit/streaming_client.py [hunk 4]
```diff
@@ -136,32 +138,36 @@ def socket_timeout(self, timeout: typing.Union[None, float]):
...
 
     def start_with_external_source(self,
+                                   ws_url: str,
                                    response_types: ResponseType = ResponseType.Transcript) -> typing.Iterator[typing.Dict]:
         """
         Start a WebSocket session and get back speech recognition responses from the server, provided that the media
         is coming from an external source.
         The media source should be configured when booking the session, via Verbit's Ordering API (see README.md)
 
+        :param ws_url: websocket url to use.
```
Comment (EliAsor) [inline verbit/streaming_client.py:165]: same here

File: verbit/streaming_client.py [hunk 5]
```diff
@@ -215,6 +222,7 @@ def _connect_and_start(self,
         Start a WebSocket session and get back speech recognition responses from the server.
         Media may be provided via the `media_generator` parameter or via an external source (see README.md)
 
+        :param ws_url: web socket url to use.
```
Comment (EliAsor) [inline verbit/streaming_client.py:225]: same here

File: verbit/streaming_client.py [hunk 6]
```diff
@@ -263,12 +274,13 @@ def _connect_websocket(self, media_config: MediaConfig, response_types: Response
         Which is linked and documented in tenacity:
         https://tenacity.readthedocs.io/en/latest/api.html#wait-functions
 
+        :param ws_url: base ws url to use
```
Comment (EliAsor) [inline verbit/streaming_client.py:277]: why base? it's the same url passed all along the way, no?

File: verbit/streaming_client.py [hunk 7]
```diff
@@ -541,7 +553,33 @@ def _get_ws_connect_query_string(media_config: MediaConfig, response_types: Resp
...
-        return {'Authorization': f'Bearer {self._ws_access_token}'}
+
+        try:
+            auth_token = self._get_auth_token()
+        except Exception:
+            self._logger.exception(f"Failed to get auth token.")
+            raise
+
+        if not auth_token:
+            err_msg = f"Failed to get valid auth token for api_key {self._customer_token}"
```
Comment (EliAsor) [inline verbit/streaming_client.py:564]:
```suggestion
            err_msg = f"Failed to get valid auth token for customer_token: {self._customer_token}"
```

**Review by EliAsor — COMMENTED** [review]
Looks great!!
Wrote very minor comments, mostly about comments.
However, we also need to update the README.md file to reflect the new auth+connection flow.

File: examples/example_client.py (hunk 1 above)
Comment (elad-shtilerman) [reply examples/example_client.py]: done

File: verbit/streaming_client.py (hunk 2 above)
Comment (elad-shtilerman) [reply verbit/streaming_client.py]: done

File: verbit/streaming_client.py (hunk 3 above)
Comment (elad-shtilerman) [reply verbit/streaming_client.py]: done with the suggested phrasing, also made other occurrences uniform.

File: verbit/streaming_client.py (hunk 4 above)
Comment (elad-shtilerman) [reply verbit/streaming_client.py]: done

File: verbit/streaming_client.py (hunk 5 above)
Comment (elad-shtilerman) [reply verbit/streaming_client.py]: done

File: verbit/streaming_client.py (hunk 6 above)
Comment (elad-shtilerman) [reply verbit/streaming_client.py]: you are right, done

File: verbit/streaming_client.py (hunk 7 above)
Comment (elad-shtilerman) [reply verbit/streaming_client.py]: done

**Review by EliAsor — COMMENTED** [review]
Looks great! so after the README update I will approve

File: README.md [hunk 8]
```diff
@@ -20,22 +20,25 @@ To install this package run:
...
 
+### Verbit API key
+To access the Verbit's Streaming Speech Recognition services first an API key (customer token) should be created by following the documentation in the auth section of the [Ordering API docs](https://platform.verbit.co/api_docs).  
+
 ### Ordering API
-In order to use Verbit's Streaming Speech Recognition services, you must place an order using Verbit's Ordering API. Your request to the Ordering API must specify that the desired input and output schemes are streaming through a WebSocket. Upon successful placement of the order, you will be issued an authentication token which can be used to initiate a WebSocket connection.
+In order to use Verbit's Streaming Speech Recognition services, you must place an order using Verbit's Ordering API. Your request to the Ordering API must specify that the desired input and/or output schemes are streaming through a WebSocket. Upon successful placement of the order, you will be issued a webscoket url composed of the base streaming api url with a query string parameters that include the order's authentication token. The url together with the customer token in the headers will be used to initiate a WebSocket connection.
```
[the remark this answers is not in the document]
Comment (elad-shtilerman) [reply README.md]: done

File: README.md [hunk 9]
```diff
@@ -20,22 +20,25 @@ To install this package run:
...
 
 These two APIs and their respective SDKs are separated on purpose because placing orders to Verbit's Transcription services does not necessarily imply media streaming (you might want to upload a file instead).
 Also, the services which operate order placement and the actual streaming of media are commonly distinct, therefore we find it useful to separate the SDKs to allow maximal flexibility for our customers.
 
-Please refer to our documentation here: [Ordering API](https://platform.verbit.co/api_docs).
+For further details regarding the ordering api, please refer to the documentation here: [Ordering API](https://platform.verbit.co/api_docs).
```
[the remark this answers is not in the document]
Comment (elad-shtilerman) [reply README.md]: done

File: README.md [hunk 10]
```diff
@@ -48,7 +51,7 @@ The `WebSocketStreamingClient` will use your generator as input, iterating it an
 
 #### Example
 
-The following example reads audio from a WAV file and streams it to the Speech Recognition Service:
+The following example reads audio from a WAV file and streams it to the Speech Recognition Service (Note: The example assumes that the customer token and websocket url have been obtained via their respective api calls):
```
[the remark this answers is not in the document]
Comment (elad-shtilerman) [reply README.md]: done

**Review by EliAsor — DISMISSED** [review]

**Review by EliAsor — APPROVED** [review]

**Review by EliAsor — APPROVED** [review]

**Pull request merged**